### PR TITLE
fix: [PL-27799]: hard delete secrets, variables, connectors in account scope when an account gets deleted

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
Whenever an account gets deleted, all the account scoped entities like secrets, connectors, variables will listen to the account deletion event and will delete the corresponding entities

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36883)

File: 120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java [hunk 1]
```diff
@@ -98,6 +102,25 @@ private boolean processSecretEntityChangeEvent(EntityChangeDTO entityChangeDTO,
...
 
+  private boolean processAccountChangeEvent(Message message) {
+    AccountEntityChangeDTO accountEntityChangeDTO;
+    try {
+      accountEntityChangeDTO = AccountEntityChangeDTO.parseFrom(message.getMessage().getData());
+    } catch (InvalidProtocolBufferException e) {
+      throw new InvalidRequestException(
+          String.format("Exception in unpacking EntityChangeDTO for key %s", message.getId()), e);
+    }
+    String action = message.getMessage().getMetadataMap().get(ACTION);
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java:113]: Check for null cases here

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Review by gn-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)* status: 2 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java` line 105 at r2](https://reviewable.io/reviews/harness/harness-core/36883#-NAg5ivMAXAIf6d7wqL8:-NAg5ivMAXAIf6d7wqL9:b-y5zg9m) ([raw file](https://github.com/harness/harness-core/blob/923a9f3dc5ef66cf47ba31de034a4e8f446b84aa/120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java#L105)):*
> ```Java
>   }
> 
>   private boolean processAccountChangeEvent(Message message) {
> ```

First thing to be done here is to check for delete then and then if needed read AccountEntityChangeDTO.

String action = message.getMessage().getMetadataMap().get(ACTION);
if (action == null || !DELETE_ACTION.equals(actions)) {
  return true;
}

AccountEntityChangeDTO accountEntityChangeDTO;
try {
    accountEntityChangeDTO = AccountEntityChangeDTO.parseFrom(message.getMessage().getData());
} catch (InvalidProtocolBufferException e) {
  throw new InvalidRequestException(
   String.format("Exception in unpacking EntityChangeDTO for key %s", message.getId()), e);
}

return processAccountDeleteEvent(accountEntityChangeDTO);
___
*[`120-ng-manager/src/test/java/io/harness/ng/core/outbox/SecretEntityCRUDStreamListenerTest.java` line 64 at r2](https://reviewable.io/reviews/harness/harness-core/36883#-NAg6fHe-cGXYQs-4cDk:-NAg6fHe-cGXYQs-4cDl:b-jhidrr) ([raw file](https://github.com/harness/harness-core/blob/923a9f3dc5ef66cf47ba31de034a4e8f446b84aa/120-ng-manager/src/test/java/io/harness/ng/core/outbox/SecretEntityCRUDStreamListenerTest.java#L64)):*
> ```Java
>   @Owner(developers = VIKAS_M)
>   @Category(UnitTests.class)
>   public void testAccountDeleteEvent() {
> ```

test null handling too



_Code quote:_
```Java
testAccountDeleteEvent
```

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)* status: 2 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java` line 105 at r2](https://reviewable.io/reviews/harness/harness-core/36883#-NAg5ivMAXAIf6d7wqL8:-NAgcMs_1P647aYJhny4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/923a9f3dc5ef66cf47ba31de034a4e8f446b84aa/120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java#L105)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

First thing to be done here is to check for delete then and then if needed read AccountEntityChangeDTO.

String action = message.getMessage().getMetadataMap().get(ACTION);
if (action == null || !DELETE_ACTION.equals(actions)) {
  return true;
}

AccountEntityChangeDTO accountEntityChangeDTO;
try {
    accountEntityChangeDTO = AccountEntityChangeDTO.parseFrom(message.getMessage().getData());
} catch (InvalidProtocolBufferException e) {
  throw new InvalidRequestException(
   String.format("Exception in unpacking EntityChangeDTO for key %s", message.getId()), e);
}

return processAccountDeleteEvent(accountEntityChangeDTO);
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java` line 113 at r2](https://reviewable.io/reviews/harness/harness-core/36883#-NAdQA4-xJg22L8s7xYo-r2-113:-NAgcNptEDfghFVKfqn2:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/923a9f3dc5ef66cf47ba31de034a4e8f446b84aa/120-ng-manager/src/main/java/io/harness/ng/core/event/SecretEntityCRUDStreamListener.java#L113)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Check for null cases here
</blockquote></details>

Done.
___
*[`120-ng-manager/src/test/java/io/harness/ng/core/outbox/SecretEntityCRUDStreamListenerTest.java` line 64 at r2](https://reviewable.io/reviews/harness/harness-core/36883#-NAg6fHe-cGXYQs-4cDk:-NAgcP1j56Nfsk0rmxmV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/923a9f3dc5ef66cf47ba31de034a4e8f446b84aa/120-ng-manager/src/test/java/io/harness/ng/core/outbox/SecretEntityCRUDStreamListenerTest.java#L64)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

test null handling too
</blockquote></details>

Done.

**Review by gn-harness — COMMENTED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)* status: 2 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained.

**Review by gn-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by gn-harness — APPROVED** [review]
> … gets delete
Please update the PR description and title to reflect what all entities are deleted.

> Reviewed 2 of 2 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883#-:-NAgjv168h40AhJ2gZxZ:bb3hdt3)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by VikasMaddukuriHarness** [comment]
trigger ti2

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, 2 of 2 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)* status: 1 change requests. 1. unresolved discussions.

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, 2 of 2 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36883)* status: :shipit: complete! ✓

___
*[`120-ng-manager/modules/ng-variables/src/main/java/io/harness/ng/core/event/VariableEntityCRUDStreamListener.java` line 66 at r5](https://reviewable.io/reviews/harness/harness-core/36883#-NAgh07w6uYq2HnxF1le-r5-66:-NAi64674pD6VJ6iGJQy:b-jnmpwy) ([raw file](https://github.com/harness/harness-core/blob/f2db23168986228c7465ce8ba1ffa0ab0dc67493/120-ng-manager/modules/ng-variables/src/main/java/io/harness/ng/core/event/VariableEntityCRUDStreamListener.java#L66)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Should we instead consider switch case for Delete, Update and Create Actions, with noop in Create and update for now?
</blockquote></details>

Lets consider this for future after analysing the maintainability of code. Approving

**Pull request merged**